coalib/misc/Caching.py

```diff
@@ -0,0 +1,142 @@
+import os
+import calendar
+import time
+import appdirs
+import pickle
```

Comment:
On CTRL+C pickle ignores it and writes everything to file. If you kill the process its mostly corrupted^^
In this case we just need to catch `UnpicklingError`^^